Sporadic Error 5xx from github

### Required prerequisites

- [x] Consult the [security policy](https://github.com/NVIDIA/cuda-quantum/security/policy). If reporting a security vulnerability, do not report the bug using this form. Use the process described in the policy to report the issue.
- [x] Make sure you've read the [documentation](https://nvidia.github.io/cuda-quantum/latest). Your issue may be addressed there.
- [x] Search the [issue tracker](https://github.com/NVIDIA/cuda-quantum/issues) to verify that this hasn't already been reported. +1 or comment there if it has.
- [x] If possible, make a PR with a failing test to give us a starting point to work on!

### Describe the bug

See https://github.com/NVIDIA/cuda-quantum/actions/runs/26593693410/job/78374012545

```
Run set -euo pipefail
push to NGC? true
Stitching release image with tag: nvcr.io/nvidia/nightly/cuda-quantum:cu12-latest-base
References:  ghcr.io/nvidia/cuda-quantum@sha256:be448cfee1559a3858d3d772d0e9afeae936bb4873944c5666bd2dbf12035f31 ghcr.io/nvidia/cuda-quantum@sha256:bf26813c8b801040801d65a4fa96fd8bc6989238d9280a4b07d9060af711d7d1
#1 [internal] pushing nvcr.io/nvidia/nightly/cuda-quantum
#1 0.000 copying sha256:bf26813c8b801040801d65a4fa96fd8bc6989238d9280a4b07d9060af711d7d1 from ghcr.io/nvidia/cuda-quantum@sha256:bf26813c8b801040801d65a4fa96fd8bc6989238d9280a4b07d9060af711d7d1 to nvcr.io/nvidia/nightly/cuda-quantum
#1 0.000 copying sha256:be448cfee1559a3858d3d772d0e9afeae936bb4873944c5666bd2dbf12035f31 from ghcr.io/nvidia/cuda-quantum@sha256:be448cfee1559a3858d3d772d0e9afeae936bb4873944c5666bd2dbf12035f31 to nvcr.io/nvidia/nightly/cuda-quantum
#1 ERROR: copy sha256:be448cfee1559a3858d3d772d0e9afeae936bb4873944c5666bd2dbf12035f31 from ghcr.io/nvidia/cuda-quantum@sha256:be448cfee1559a3858d3d772d0e9afeae936bb4873944c5666bd2dbf12035f31 to nvcr.io/nvidia/nightly/cuda-quantum:cu12-latest-base: failed to authorize: failed to fetch oauth token: unexpected status from GET request to https://nvcr.io/proxy_auth?scope=repository%3Anvidia%2Fnightly%2Fcuda-quantum%3Apull&scope=repository%3Anvidia%2Fnightly%2Fcuda-quantum%3Apull%2Cpush: 500 Internal Server Error
------
 > [internal] pushing nvcr.io/nvidia/nightly/cuda-quantum:
0.000 copying sha256:bf26813c8b801040801d65a4fa96fd8bc6989238d9280a4b07d9060af711d7d1 from ghcr.io/nvidia/cuda-quantum@sha256:bf26813c8b801040801d65a4fa96fd8bc6989238d9280a4b07d9060af711d7d1 to nvcr.io/nvidia/nightly/cuda-quantum
0.000 copying sha256:be448cfee1559a3858d3d772d0e9afeae936bb4873944c5666bd2dbf12035f31 from ghcr.io/nvidia/cuda-quantum@sha256:be448cfee1559a3858d3d772d0e9afeae936bb4873944c5666bd2dbf12035f31 to nvcr.io/nvidia/nightly/cuda-quantum
------
ERROR: copy sha256:be448cfee1559a3858d3d772d0e9afeae936bb4873944c5666bd2dbf12035f31 from ghcr.io/nvidia/cuda-quantum@sha256:be448cfee1559a3858d3d772d0e9afeae936bb4873944c5666bd2dbf12035f31 to nvcr.io/nvidia/nightly/cuda-quantum:cu12-latest-base: failed to authorize: failed to fetch oauth token: unexpected status from GET request to https://nvcr.io/proxy_auth?scope=repository%3Anvidia%2Fnightly%2Fcuda-quantum%3Apull&scope=repository%3Anvidia%2Fnightly%2Fcuda-quantum%3Apull%2Cpush: 500 Internal Server Error
Error: Process completed with exit code 1.
```

### Steps to reproduce the bug

Run CI

### Expected behavior

Gracefully handle 500

### Is this a regression? If it is, put the last known working version (or commit) here.

Not a regression

### Environment

- **CUDA-Q version**: 
- **Python version**: 
- **C++ compiler**: 
- **Operating system**: 


### Suggestions

_No response_

--- a/.github/workflows/deployments.yml
+++ b/.github/workflows/deployments.yml
@@ -602,6 +602,17 @@ jobs:
           set -euo pipefail
           DIGESTS_JSON='${{ steps.metadata.outputs.digests_json }}'
 
+          # Retry registry-hitting commands; nvcr.io/ghcr.io occasionally
+          # return transient 5xx from their auth/proxy endpoints.
+          retry() {
+            for n in 1 2 3 4 5; do
+              "$@" && return 0
+              [ "$n" -eq 5 ] && return 1
+              echo "Command '$*' failed (attempt $n/5); retrying in $((10*n))s..." >&2
+              sleep $((10*n))
+            done
+          }
+
           cuda="${{ matrix.cuda }}"
           cuda_major="${cuda%%.*}"
 
@@ -624,8 +635,8 @@ jobs:
           echo "Stitching debug image with tag: $stitched_tag"
           echo "References: $refs"
 
-          docker buildx imagetools create --tag "$stitched_tag" $refs
-          digest=$(docker buildx imagetools inspect "$stitched_tag" | awk '/^Digest:/ { print $2 }')
+          retry docker buildx imagetools create --tag "$stitched_tag" $refs
+          digest=$(retry docker buildx imagetools inspect "$stitched_tag" | awk '/^Digest:/ { print $2 }')
           stitched_output=$(cat stitched_digests.json 2>/dev/null || echo '{}')
           stitched_output=$(echo "$stitched_output" | jq --arg key "dev-cu${cuda_major}" --arg val "$registry@$digest" '. + {($key): $val}')
           echo "$stitched_output" | tee stitched_digests.json
@@ -634,6 +645,17 @@ jobs:
         run: |
           set -euo pipefail
           DIGESTS_JSON='${{ steps.metadata.outputs.digests_json }}'
+
+          # Retry registry-hitting commands; nvcr.io/ghcr.io occasionally
+          # return transient 5xx from their auth/proxy endpoints.
+          retry() {
+            for n in 1 2 3 4 5; do
+              "$@" && return 0
+              [ "$n" -eq 5 ] && return 1
+              echo "Command '$*' failed (attempt $n/5); retrying in $((10*n))s..." >&2
+              sleep $((10*n))
+            done
+          }
 
           cuda="${{ matrix.cuda }}"
           cuda_major="${cuda%%.*}"
@@ -668,7 +690,7 @@ jobs:
             image_tag+=pr-${pr_number}
           elif ${{ github.ref_type == 'branch' && github.ref_name == 'main' }}; then
             image_tag+="latest"
-          elif "$is_versioned" == "true"; then
+          elif [[ "$is_versioned" == "true" ]]; then
             image_tag+=`echo ${{ github.ref_name }} | egrep -o "([0-9]{1,}\.)+[0-9]{1,}"`
           else
             image_tag+=`echo ${{ github.ref_name }} | tr '/' '-'`
@@ -682,8 +704,8 @@ jobs:
           echo "Stitching release image with tag: $stitched_tag"
           echo "References: $refs"
 
-          docker buildx imagetools create --tag "$stitched_tag" $refs
-          digest=$(docker buildx imagetools inspect "$stitched_tag" | awk '/^Digest:/ { print $2 }')
+          retry docker buildx imagetools create --tag "$stitched_tag" $refs
+          digest=$(retry docker buildx imagetools inspect "$stitched_tag" | awk '/^Digest:/ { print $2 }')
           stitched_output=$(cat stitched_digests.json 2>/dev/null || echo '{}')
           stitched_output=$(echo "$stitched_output" | jq --arg key "base-cu${cuda_major}" --arg val "$registry@$digest" '. + {($key): $val}')
           echo "$stitched_output" | tee stitched_digests.json

